### PR TITLE
DROOLS-4407: [DMN Designer] Search feature - Highlight the current result with colors that follow the PF standard

### PR DESCRIPTION
**Pull request by danielzhe**
This PR is part of an ensemble and also covers [DROOLS-4406](https://issues.jboss.org/projects/DROOLS/issues/DROOLS-4406).

https://github.com/kiegroup/drools-wb/pull/1214
https://github.com/kiegroup/appformer/pull/767
https://github.com/kiegroup/kie-wb-common/pull/2815

### CODE REVIEW

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/themes/GridRendererTheme.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
  */
 public interface GridRendererTheme {
 
+    String HIGHLIGHTED_CELL_BACKGROUND = "#F9D67A";
+
+    String HIGHLIGHTED_CELL_STROKE = "#F5C12E";
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/themes/GridRendererTheme.java]: I think we should move these two constants into:
`org.uberfire.ext.wires.core.grids.client.widget.grid.renderers.themes.impl.KIEColours`

**Comment by danielzhe** [comment]
@jomarko Changes applied.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRenderer.java [hunk 2]
```diff
@@ -439,4 +451,124 @@ public boolean onGroupingToggle(double cellX,
...
+            int firstRowOfMergedBlock = getFirstRowIndexOfMergedBlock(model, visibleRowIndex);
+            r.setY(rendererHelper.getRowOffset(firstRowOfMergedBlock));
+        }
+    }
+
+    int getFirstRowIndexOfMergedBlock(final GridData model, final int rowIndex) {
+
+        int currentIndex = rowIndex;
+        GridCell<?> cell = model.getCell(rowIndex, getHighlightCellColumnIndex());
+        while (cell.getMergedCellCount() == 0) {
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRenderer.java]: I think we should have the same condition as we have in `getMergedCellsCount`:
```java
cell.getMergedCellCount() == 0 && currentIndex >= 0
```

Comment (jomarko) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRenderer.java]: @danielzhe hello, please let me know what do you think about this comment.

Comment (danielzhe) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRenderer.java]: Changes applied.

**Comment by karreiro** [comment]
@danielzhe I think Sonar is failing because the code coverage is less than 80%:

Screen Shot 2019-08-29 at 17 34 08

Screen Shot 2019-08-29 at 17 34 19

**Comment by danielzhe** [comment]
@karreiro @manstis Finally we can merge this.

**Comment by karreiro** [comment]
Thank you, @danielzhe :-)

**Pull request merged**